module/zfs/spa_misc.c

```diff
@@ -1770,6 +1791,35 @@ spa_deadman_synctime(spa_t *spa)
 	return (spa->spa_deadman_synctime);
 }
 
+uint64_t
+spa_deadman_ziotime(spa_t *spa)
+{
+	return (spa->spa_deadman_ziotime);
+}
+
+uint64_t
+spa_get_deadman_failmode(spa_t *spa)
+{
+	return (spa->spa_deadman_failmode);
+}
+
+int
+spa_set_deadman_failmode(spa_t *spa, const char *failmode)
+{
+	int error = 0;
+
+	if (strcmp(failmode, "wait") == 0)
+		spa->spa_deadman_failmode = ZIO_FAILURE_MODE_WAIT;
```

Comment:
Maybe the default should be to set to ```ZIO_FAILURE_MODE_WAIT``` and then this can become a ```void```-returning function and the return value check in ```spa_add()``` can be removed.

Reply:
Sure.